ONSdigital/rm-survey-service
# Add delete endpoint

# What and why?

This PR adds a DELETE /survey/<survey-id> endpoint to the service so we can delete any surveys.  For this, we trust that any validation is done beforehand by other services so if this is getting hit then it's safe to delete the record.

Additionally, this also adds cascade on deletes to a few of the foreign keys as the classifier records are related to the survey, so if the survey doesn't exist, then the related classifier records should also be removed as well.

# How to test?

Deploy to your environment and hit the endpoint with a valid id.  The record and classifiers should be deleted.  Additionally, wrongly formatted uuids should return a 400 and trying to delete a record that doesn't exist should return a 404.


# Trello
